services/management/dbaas/psmdb_cluster_service.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// PSMDBClusterService implements PSMDBClusterServer methods.
+type PSMDBClusterService struct {
+	db               *reform.DB
+	l                *logrus.Entry
+	controllerClient dbaasClient
+}
+
+// NewPSMDBClusterService creates PSMDB Service.
+func NewPSMDBClusterService(db *reform.DB, client dbaasClient) dbaasv1beta1.PSMDBClusterServer {
+	l := logrus.WithField("component", "xtradb_cluster")
+	return &PSMDBClusterService{db: db, l: l, controllerClient: client}
+}
+
+// ListPSMDBClusters returns a list of all PSMDB clusters.
+func (s PSMDBClusterService) ListPSMDBClusters(ctx context.Context, req *dbaasv1beta1.ListPSMDBClustersRequest) (*dbaasv1beta1.ListPSMDBClustersResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(s.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	in := dbaascontrollerv1beta1.ListPSMDBClustersRequest{
+		KubeAuth: &dbaascontrollerv1beta1.KubeAuth{
+			Kubeconfig: kubernetesCluster.KubeConfig,
+		},
+	}
+
+	out, err := s.controllerClient.ListPSMDBClusters(ctx, &in)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Can't get list of PSMDB clusters: %s", err.Error())
+	}
+
+	clusters := make([]*dbaasv1beta1.ListPSMDBClustersResponse_Cluster, len(out.Clusters))
+	for i, c := range out.Clusters {
+		cluster := dbaasv1beta1.ListPSMDBClustersResponse_Cluster{
+			Name: c.Name,
+			Params: &dbaasv1beta1.PSMDBClusterParams{
+				ClusterSize: c.Params.ClusterSize,
+				Replicaset: &dbaasv1beta1.PSMDBClusterParams_ReplicaSet{
+					ComputeResources: &dbaasv1beta1.ComputeResources{
+						CpuM:        c.Params.Replicaset.ComputeResources.CpuM,
+						MemoryBytes: c.Params.Replicaset.ComputeResources.MemoryBytes,
+					},
+				},
+			},
+		}
+
+		clusters[i] = &cluster
+	}
+
+	return &dbaasv1beta1.ListPSMDBClustersResponse{Clusters: clusters}, nil
+}
+
+// ShowPSMDBCluster returns a PSMDB cluster by name.
+func (s PSMDBClusterService) ShowPSMDBCluster(ctx context.Context, req *dbaasv1beta1.ShowPSMDBClusterRequest) (*dbaasv1beta1.ShowPSMDBClusterResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(s.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: implement on dbaas-controller side:
+	// 1. Get psmdb host and status
+	//  - Ex.: kubectl get -o=json PerconaServerMongoDB/<cluster_name>
+	// 2. Get root password:
+	//   - Ex.: kubectl get secret my-cluster-name-secrets -o json  | jq -r ".data.MONGODB_USER_ADMIN_PASSWORD" | base64 -d
+	_ = kubernetesCluster
+	resp := dbaasv1beta1.ShowPSMDBClusterResponse{
+		Name:     req.Name,
+		Username: "userAdmin",
+		Password: "userAdmin123456",
+		Host:     fmt.Sprintf("%s-rs0.default.svc.cluster.local", req.Name),
+		Port:     3306,
```

Comment:
Maybe we need to return replicaSet here as well, as it is not very obvious that it is 'rs0'